ClassMoodApp/static/tsc/controllers/LiveViewController.ts

```diff
@@ -0,0 +1,138 @@
+                    {
+                        this.getGauge(this);
+                    }
+                }
+            )
+        }
+
+        private getLectureId(): void {
+            this.lectureId = parseInt($("#lectureId").attr('lecture-id'), 10);
+        }
+
+        private votePace(newPace: number): void {
+            this.gauge.pace_num = newPace;
+            this.postCurrentGauge();
+        }
+
+        private voteDepth(newDepth: number): void {
+            this.gauge.depth_num = newDepth;
+            this.postCurrentGauge();
+        }
+
+        private getGauge(self): void {
+            self.$http.get(`/live_lecture/gauge/get/${self.lectureId}`)
+            .success(function(response: any) {
+                self.gauge.pace_num = response.results[0];
+                if (self.gauge.pace_num == -1)
+                {
+                    var pace = $("#pace1");
+                    $("#pace1").prop('checked', true);
+                    $("#pace1").parent().addClass('active');
+                }
+                else if (self.gauge.pace_num == 1)
+                {
+                    $("#pace3").prop('checked', true);
+                    $("#pace3").parent().addClass('active');
+                }
+                else
+                {
+                    $("#pace2").prop('checked', true);
+                    $("#pace2").parent().addClass('active');
+                }
+                self.gauge.depth_num = response.results[1];
+                if (self.gauge.depth_num == -1)
+                {
+                    $("#depth1").prop('checked', true);
+                    $("#depth1").parent().addClass('active');
+                }
+                else if (self.gauge.depth_num == 1)
+                {
+                    $("#depth3").prop('checked', true);
+                    $("#depth3").parent().addClass('active');
+                }
+                else
+                {
+                    $("#depth2").prop('checked', true);
+                    $("#depth2").parent().addClass('active');
+                }
+            })
+            .error(function(response: any) {
+            })
+        }
+
+        private postCurrentGauge(): void {
+            this.$http.post('/live_lecture/gauge/update', JSON.stringify({lect_id: this.lectureId, pace_num: this.gauge.pace_num, depth_num: this.gauge.depth_num}))
+            .success(function(data, status, headers, config) {
+            })
+            .error(function(data, status, header, config) {
+                console.log("ERROR while posting current gauge");
+            })
+        }
+
+        private gaugePollPace(self): void {
+            this.$http.get(`/live_lecture/pace/get/${this.lectureId}`)
+            .success(function(response: number) {
+                self.gauge.pace_num = response;
+                $('#paceSlider').get(0).MaterialSlider.change(self.gauge.pace_num);
+                self.$timeout(function() {self.gaugePollPace(self)}, 5000);
```

Comment:
Angular's way of doing repeated timed function calls. Basically says to call the gaugePollPace() function every 5 seconds to poll the pace from the server.